lib/open_food_network/feature_toggle.rb

```diff
@@ -1,21 +1,78 @@
 module OpenFoodNetwork
+  # This feature toggles implementation provides two mechanisms to conditionally enable features.
+  #
+  # You can provide an ENV var with the prefix `OFN_FEATURE_` and query it using the
+  # `ApplicationHelper#feature?` helper method. For instance, providing the ENV var
+  # `OFN_FEATURE_NEW_SHINNY_FEATURE` you could then query it from view as follows:
+  #
+  #   - if feature? :new_shinny_feature
+  #     = render "new_shinny_feature"
+  #
+  # Alternatively, you can choose which users have the feature toggled on. To do that you need to
+  # register the feature and its users from an initializer like:
+  #
+  #   require 'open_food_network/feature_toggle'
+  #   OpenFoodNetwork::FeatureToggle.enable(:new_shinny_feature, ['ofn@example.com'])
+  #
+  # It's although advisable to read the emails list from an ENV Var.
+  #
+  # You can then check it from a view like:
+  #
+  #   - if feature? :new_shinny_feature, spree_current_user
+  #     = render "new_shinny_feature"
+  #
   class FeatureToggle
-    def self.enabled?(feature_name)
-      true?(env_variable_value(feature_name))
+    def self.enabled?(feature_name, user = nil)
+      new.enabled?(feature_name, user)
+    end
+
+    def self.enable(feature_name, user_emails)
+      Thread.current[:features] ||= {}
+      Thread.current[:features][feature_name] = Feature.new(:foo, user_emails)
+    end
+
+    def initialize
+      @features = Thread.current[:features]
+    end
```

Comment:
I guess I'm a little wary of using threads in this way... it looks like we were previously just used ENV variables. What was the reason for switching from that? Are there any potential side effects with storing `:features` on the thread like this, or potential bugs if there are things running on multiple threads?

Reply:
The problem is that the set of features needs to be persisted across requests. It's something that needs to be loaded in memory at all times so we can access it when doing `FeatureToggle.enabled?(:foo, user)`. A common way to do this is storing the state as a thread-local var (sort of, I just found there's fibers involved). 

In the case of a multi-threaded app server (thinking about Puma here), I'm assuming all threads would have the same state since they would all go through the initializer and no writes to the state would happen at runtime. There's no state shared between threads, which would bring chaos.

Reply:
Got it, that makes sense to me now. Flipper does look nice if we ever want to do this more extensively.

Reply:
Yes, and checking quickly Rollout, I saw it needs Redis apparently so can't use it just yet unless we introduce yet another dependency in the infrastructure :cold_sweat: